lib/Service/SnoozeService.php

```diff
@@ -112,7 +202,28 @@ public function snoozeThread(Message $selectedMessage, int $unixTimestamp): void
 			$snooze = new MessageSnooze();
 			$snooze->setMessageId($message['messageId']);
 			$snooze->setSnoozedUntil($unixTimestamp);
-			$this->messageSnoozeMapper->insert($snooze);
+			try {
+				$this->messageSnoozeMapper->insert($snooze);
+			} catch(Exception $e) {
+				$this->messageSnoozeMapper->deleteByMessageId($message['messageId']);
+				$this->messageSnoozeMapper->insert($snooze);
+			}
+		}
+	}
+
+	/**
+	 * Removes DB entry for the messages
+	 *
+	 * @param Message $selectedMessage
+	 * @return void
+	 */
+	public function unSnoozeThreadDB(Message $selectedMessage): void {
+		$messages = $this->threadMapper->findMessageIdsByThreadRoot(
+			$selectedMessage->getThreadRootId()
+		);
+
+		foreach ($messages as $message) {
+			$this->messageSnoozeMapper->deleteByMessageId($message['messageId']);
 		}
```

Comment:
perf: would be faster and atomic if this is done in one query by the db

Reply:
Changed `deleteByMessageId()` to take array of ids and query db by where->in